crates/ruff_server/src/server.rs

```diff
@@ -236,3 +237,69 @@ impl Server {
         }
     }
 }
+
+bitflags::bitflags! {
+    #[derive(Clone, Copy, Debug, PartialEq, Eq)]
+    struct AvailableCodeActions: u8 {
+        const QUICK_FIX = 0b0000_0001;
+        const SOURCE_FIX_ALL = 0b0000_0010;
+        const SOURCE_ORGANIZE_IMPORTS = 0b0000_0100;
+    }
+}
+
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub(crate) enum SupportedCodeActionKind {
+    QuickFix,
+    SourceFixAll,
+    SourceFixAllRuff,
+    #[allow(dead_code)] // TODO: remove
+    SourceOrganizeImports,
+    #[allow(dead_code)] // TODO: remove
+    SourceOrganizeImportsRuff,
+}
+
+impl SupportedCodeActionKind {
+    fn kind(self) -> CodeActionKind {
+        match self {
+            Self::QuickFix => CodeActionKind::QUICKFIX,
+            Self::SourceFixAll => CodeActionKind::SOURCE_FIX_ALL,
+            Self::SourceFixAllRuff => crate::SOURCE_FIX_ALL_RUFF,
+            Self::SourceOrganizeImports => CodeActionKind::SOURCE_ORGANIZE_IMPORTS,
+            Self::SourceOrganizeImportsRuff => crate::SOURCE_ORGANIZE_IMPORTS_RUFF,
+        }
+    }
+
+    fn makes_available(self) -> AvailableCodeActions {
```

Comment:
We take `self` by value here because `Self` is `Copy`-able and clippy complains if you take it by reference. It's not meant to be a conversion method, but maybe that needs to be clarified.